src/State/Provider/DeserializeProvider.php

```diff
@@ -36,6 +37,7 @@ public function __construct(
         private readonly ?ProviderInterface $decorated,
         private readonly SerializerInterface $serializer,
         private readonly SerializerContextBuilderInterface $serializerContextBuilder,
+        private readonly ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory,
```

Comment:
It bothers me that we need to inject this but I guess there's no other solution unless we only use reflection?

Reply:
Yes, if we want to use a short name, we need this service.

I would leave only a reflection, for simplicity. However, I can't assess how well it meets the general concept of the API Platform.